jetson/science_bridge/src/__main__.py

```diff
@@ -201,156 +203,71 @@ def heater_transmit(self, channel, msg):
+        message = self.format_mosfet_message(translated_device, int(struct.enable))
+        self.uart_send(message)
         print("Mosfet Received")
         pass
 
-    def nav_status(self, channel, msg):  # TODO - fix make better
-        print("Received nav req")
-        # Off, Done, Else
-
-        struct = NavStatus.decode(msg)
-        message = "$Mosfet,{device},{enable}"
-
-        # Off = Blue
-        if struct.nav_state_name == "Off":
-            print("navstatus off")
-            blue = message.format(device=Mosfet_devices.BLUE_LED.value, enable=1)
-            blue = self.add_padding(blue)
-            self.ser.write(bytes(blue, encoding='utf-8'))
-            prev = Auton_state.OFF
-        # Done = Flashing green
-        elif struct.nav_state_name == "Done":
-            print("navstatus Done")
-            # Flashing by turning on and off for 1 second intervals
-
-            NUMBER_OF_LED_BLINKS = 6
-
-            for i in range(NUMBER_OF_LED_BLINKS):
-                green_on = message.format(device=Mosfet_devices.GREEN_LED.value,
-                                          enable=1)
-                green_on = self.add_padding(green_on)
-                self.ser.write(bytes(green_on, encoding='utf-8'))
-                time.sleep(1)
-                green_off = message.format(device=Mosfet_devices.GREEN_LED.value,
-                                           enable=0)
-                green_off = self.add_padding(green_off)
-                self.ser.write(bytes(green_off, encoding='utf-8'))
-                time.sleep(1)
-                prev = Auton_state.DONE
-        # Everytime else = Red
-        else:
-            print("navstatus else")
-            red = message.format(device=Mosfet_devices.RED_LED.value, enable=1)
-            red = self.add_padding(red)
-            self.ser.write(bytes(red, encoding='utf-8'))
-            prev = Auton_state.ELSE
-        time.sleep(1)
-        # Green should be in a finished state so no need to turn it off
-        if (prev != Auton_state.OFF):
-            green_off = message.format(device=Mosfet_devices.GREEN_LED.value, enable=0)
-            green_off = self.add_padding(green_off)
-            self.ser.write(bytes(green_off, encoding='utf-8'))
-        if (prev != Auton_state.ELSE):
-            red_off = message.format(device=Mosfet_devices.RED_LED.value, enable=0)
-            red_off = self.add_padding(red_off)
-            self.ser.write(bytes(red_off, encoding='utf-8'))
+    def auton_led(self, channel, msg):  # TODO - fix make better
+        # Off, Done, On
+        struct = AutonLed.decode(msg)
+
+        requested_state = LED_state.NONE
+
+        if struct.color == "Red":
+            requested_state = LED_state.RED
+        elif struct.color == "Green":
+            requested_state = LED_state.GREEN
+        elif struct.color == "Blue":
+            requested_state = LED_state.BLUE
+
+        if self.previous_led_state == requested_state:
```

Comment:
is teleop constantly spamming the nav state changes? i feel like it'll be at lot cleaner for this state change logic to be handled on their part since it's already done for the gui?

Reply:
i'm not sure if teleop constantly spams nav state changes, @jnnanni do you?

Reply:
he said it doesn't, does the esw code handle that properly then?

Reply:
I talked to Guthrie directly about it on slack, should be fine.